### PR TITLE
UI: Some settings changes

### PR DESCRIPTION
**Pull request by cg2121**
### Description

Changes:
- Remove the apply button (the settings are saved anyway when the dialog is closed, so is it necessary?)
- Close settings when escape is pressed (convenience for users)
- Users can still set the esc button as a hotkey

### Motivation and Context
Streamlining settings.

### How Has This Been Tested?
Changed settings. Settings saved correctly.

### Types of changes
- New feature (non-breaking change which adds functionality)
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
I think the apply button and confirmation are important. Without them, there is no way to discard changes. I often go in and check different settings, and then don't want to save my changes so I just close the window and discard them. I think the only part of these changes I like is the additional of the ESC key, but I am curious what others think on this as my workflow might just be odd.

**Comment by cg2121** [comment]
There is a cancel button if you want to discard changes.

**Comment by notr1ch** [comment]
The settings being saved when clicking X or pressing Esc seems to go against the common UI pattern of that action being equivalent to Cancel.

Microsoft's Windows UX guide has a good reference for property pages (which is the closest I could find for how our Settings window works). It's definitely worth a read to ensure we match user expectations. https://docs.microsoft.com/en-us/windows/win32/uxguide/win-property-win

TLDR:
- Don't confirm if users discard their changes. (Exception: If the property window has settings that require significant effort to set and the user has made changes, you may display a confirmation if the user clicks the Close button on the title bar. The reason is that some users mistakenly assume that the Close button on the title bar has the same effect as the OK button.)
- With the exception of the confirmation message, make sure the Close button on the title bar has the same effect as Cancel or Close.
- Provide an Apply button only if the property sheet has settings (at least one) with effects that users can evaluate in a meaningful way. Typically, Apply buttons are used when settings make visible changes. Users should be able to apply a change, evaluate the change, and make further changes based on that evaluation. If not, remove the Apply button instead of disabling it.

**Comment by cg2121** [comment]
The only major visible change I can think of in the settings is changing the theme, which is changed anyways without the apply button. Any others please comment.

At least on GNOME apps on Linux, the settings are saved when clicking the x button, as there is usually no buttons for OK or cancel.

I can still understand having the confirmation dialog. I'll give some more thought.

Feedback is always welcome.

**Comment by cg2121** [comment]
I've now changed my mind on removing the confirmation dialog.

**Comment by UserNaem** [comment]
The Apply button is there so that the user can apply the settings they've changed so far, and continue fiddling with settings without fear of losing their "good" settings. Removing this button would force the user to close and re-open the Settings window instead.

Almost all desktop applications discard changes or throw a confirmation prompt when you press Esc or close the settings dialog, this is the expected behaviour. If OBS wants to be special and save them instead, this will only cause user confusion.

**Comment by WizardCM** [comment]
I agree that the removal of Apply is unnecessary and makes the UX slightly worse.
However, I do agree with the change to close settings when escape is pressed, as long as it shows the confirmation dialog.

**Comment by unknowndomain** [comment]
Most of the voices here appear to be from Windows and Linux users, so bringing a Mac voice to the table...

First, the Escape button exit, this is a normal feature of application settings across macOS and if not implemented for everyone should at least be available for macOS users.

Second, removing the apply button...

[In the Apple HIG](https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/dialogs/):

> **Consider including an Apply button that lets the user preview changes before committing to them.** An Apply button makes sense in a modeless dialog for changes that can be applied, previewed, and undone quickly. Save dialogs and other dialogs that let users make changes that can’t be previewed easily should not include an Apply button.

This being said, the apply button is not a paradigm used within macOS very often if at all, it's typically in applications like OBS which have been developed primarily on other operating systems.

Most application and system settings in macOS do not have save or apply buttons, settings are just applied immediately, in the event it could be irreversable the UI pattern is to display a modal confirmation sheet on the window or sometimes a modal dialog window.

The only example I can find right now of Apple is self using an apply button is in System Preferences > Network and even then it doesn't comply with their own HIG as changing network settings and pressing apply here is actually acting as a save button but this is a very old part of the UI.

Does it really matter? Probably not, but it would be a step towards decluttering the UX if at least for macOS this button was removed if not everyone.

**Comment by cg2121** [comment]
Closing because of #2916 was merged and accomplished the same thing with the escape button. The Apply button isn't that big of a deal, so it can stay as is.
